src/v/archival/ntp_archiver_service.cc

```diff
@@ -410,4 +395,29 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::upload_next_candidates(
       });
 }
 
+uint64_t ntp_archiver::estimate_backlog_size(cluster::partition_manager& pm) {
+    auto last_offset = _manifest.size() ? _manifest.get_last_offset()
+                                        : model::offset(0);
+    auto opt_log = pm.log(_manifest.get_ntp());
+    if (!opt_log) {
+        return 0U;
+    }
+    auto log = dynamic_cast<storage::disk_log_impl*>(opt_log->get_impl());
+    uint64_t total_size = std::accumulate(
+      std::begin(log->segments()),
+      std::end(log->segments()),
+      0UL,
+      [last_offset](
+        uint64_t acc, const ss::lw_shared_ptr<storage::segment>& s) {
+          if (s->offsets().dirty_offset > last_offset) {
+              return acc + s->size_bytes();
+          }
+          return acc;
+      });
+    // Note: we can safely ignore the fact that the last segment is not uploaded
```

Comment:
Probably nothing to worry about right now (as the integral term is 0), but if there is persistent non-zero error (such as when there is always an unsealed segment with some data), integral term can accumulate and throw the controller off.